### PR TITLE
HPCC-28670 Create packages and attach as assets on tag

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: .github/workflows/build-assets.yml [hunk 1]
```diff
@@ -0,0 +1,210 @@
+name: Build Assets newbuild
+
+env:
+  VCPKG_BINARY_SOURCES: "clear;nuget,GitHub,read"
+  VCPKG_NUGET_REPOSITORY: "https://github.com/hpcc-systems/vcpkg"
+
+on:
+  push:
+    tags:
+      - "*"
```
Comment (GordonSmith) [inline .github/workflows/build-assets.yml:10]: * Should be removed before master upmerge

**Comment by GordonSmith** [comment]
@xwang2713 - One of the goals of this PR and feature branch is to retire Jenkins and build all the packages in GH Actions.  

This is done in the [build-assets.yml](https://github.com/hpcc-systems/HPCC-Platform/pull/16794/files#diff-1cb3dcabfd7d99ae883acdccc1e1729299d1245fb8607a102fccf3868aa05158) file, triggered on a tag.

At the moment it is only building the "full" versions (one package with all plugins enabled), we will want to tweak this so that it creates similar packages to the ones Jenkins produced (Smaller platform package with seperate ones for the select plugins).

**Review by ghalliday — COMMENTED** [review]
Looks sensible.  Will wait for Ming to look at it though.

**Comment by GordonSmith** [comment]
@xwang2713 bump

File: .github/workflows/build-assets.yml (hunk 1 above)
Comment (xwang2713) [reply .github/workflows/build-assets.yml:10]: Looks good

Comment (xwang2713) [reply .github/workflows/build-assets.yml:10]: There are two additional OS: 1) Ubuntu 22.10. Need to test if the latest fix resolves all the build issues 2) AWS Amazan Linux 2 (amzn2) . This is required for Tim's HMS application in AWS (hpccsystems playground app). It probably is not available in github action.

**Pull request merged**